nextcord/ui/modal.py

```diff
@@ -0,0 +1,378 @@
+        if self.__stopped.done():
+            return
+
+        asyncio.create_task(self._scheduled_task(interaction), name=f'discord-ui-modal-dispatch-{self.id}')
+    
+    def refresh(self, components: List[Component]):
+        # This is pretty hacky at the moment
+        # fmt: off
+        old_state: Dict[Tuple[int, str], Item] = {
+            (item.type.value, item.custom_id): item  # type: ignore
+            for item in self.children
+            if item.is_dispatchable()
+        }
+        # fmt: on
+        children: List[Item] = []
+        for component in _walk_all_components(components):
+            try:
+                older = old_state[(component.type.value, component.custom_id)]  # type: ignore
+            except (KeyError, AttributeError):
+                print("error", component)
+                children.append(_component_to_item(component))
+            else:
+                older.refresh_component(component)
+                children.append(older)
+
+        self.children = children
+    
+    def stop(self) -> None:
+        """Stops listening to interaction events from this modal.
+
+        This operation cannot be undone.
+        """
+        if not self.__stopped.done():
+            self.__stopped.set_result(False)
+
+        self.__timeout_expiry = None
+        if self.__timeout_task is not None:
+            self.__timeout_task.cancel()
+            self.__timeout_task = None
+
+        if self.__cancel_callback:
+            self.__cancel_callback(self)
+            self.__cancel_callback = None
+    
+    def is_finished(self) -> bool:
+        """:class:`bool`: Whether the modal has finished interacting."""
+        return self.__stopped.done()
+
+    def is_dispatching(self) -> bool:
+        """:class:`bool`: Whether the modal has been added for dispatching purposes."""
+        return self.__cancel_callback is not None
+
+    def is_persistent(self) -> bool:
+        """:class:`bool`: Whether the modal is set up as persistent.
+
+        A persistent modal has a set ``custom_id`` and all their components with a set ``custom_id`` and
+        a :attr:`timeout` set to ``None``.
+        """
+        return self._provided_custom_id and self.timeout is None and all(item.is_persistent() for item in self.children)
+
+    async def wait(self) -> bool:
+        """Waits until the modal has finished interacting.
+
+        A modal is considered finished when :meth:`stop` is called
+        or it times out.
+
+        Returns
+        --------
+        :class:`bool`
+            If ``True``, then the modal timed out. If ``False`` then
+            the modal finished normally.
+        """
+        return await self.__stopped
+
+class ModalStore:
+    def __init__(self, state: ConnectionState):
+        # (user_id, custom_id): Modal
+        self._modals: Dict[Tuple[int, str], Modal] = {}
+        self._state: ConnectionState = state
+
+    @property
+    def persistent_modals(self) -> Sequence[Modal]:
```

Comment:
```suggestion
    def persistent_modals(self) -> List[Modal]:
```
its known to be a list right?